go-selinux/selinux_linux.go

```diff
@@ -1181,6 +1183,102 @@ func dupSecOpt(src string) ([]string, error) {
 	return dup, nil
 }
 
+// checkGroup returns true if group's GID is in the list of GIDs gids.
+func checkGroup(group string, gids []string, lookupGroup func(string) (*user.Group, error)) bool {
+	grp, err := lookupGroup(group)
+	if err != nil {
+		return false
+	}
+
+	for _, gid := range gids {
+		if grp.Gid == gid {
+			return true
+		}
+	}
+	return false
+}
+
+// reads the seusers file: https://www.man7.org/linux/man-pages/man5/seusers.5.html
+func getSeUserFromReader(username string, gids []string, r io.Reader, lookupGroup func(string) (*user.Group, error)) (string, string, error) {
+	var defaultSeUser, defaultLevel string
+	var groupSeUser, groupLevel string
+
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		line = strings.TrimSpace(line)
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+		parts := strings.SplitN(line, ":", 3)
```

Comment:
Is the file syntactically valid if there are more than 3 parts? IMO this should reject syntactically invalid files to prevent accidentally mapping to the wrong selinux user.

Reply:
I was leaning towards continuing to avoid hard failing when an invalid line is encountered, but since this is dealing with system security I think failing fast is the safer option, added